main.tf

```diff
@@ -1,13 +1,103 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled          = module.this.enabled
+  iam_role_enabled = local.enabled && var.iam_role_enabled
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
-  }
+module "eks_iam_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.1.0"
+
+  enabled = local.iam_role_enabled
+
+  iam_source_json_url   = var.iam_source_json_url
+  iam_policy_statements = var.iam_policy_statements
+
+  context = module.this.context
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "eks_iam_role" {
+  source  = "cloudposse/eks-iam-role/aws"
+  version = "0.8.0"
+
+  enabled = local.iam_role_enabled
+
+  aws_account_number          = var.aws_account_number
+  aws_iam_policy_document     = local.iam_role_enabled ? module.eks_iam_policy.json : "{}"
+  aws_partition               = var.aws_partition
```

Comment:
Why don't we just use the [aws_partition data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition) for this?

This is probably because we are not making the `aws` provider a required one. But perhaps we should consider simplifying this module and just relying on the `aws` provider?

Reply:
Not sure. This is how we've been doing it so far. I think the data source belongs in the eks iam role module then. Perhaps we can release this initially, then put in a PR to the eks iam role module, then update this root module to use the newest version.